examples/shell/mbed/cmd_mbed_utils.cpp

```diff
@@ -0,0 +1,124 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <lib/shell/shell.h>
+
+#include <lib/core/CHIPCore.h>
+#include <lib/support/CHIPArgParser.hpp>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/TimeUtils.h>
+
+#include <inttypes.h>
+#include <stdarg.h>
+
+#include <ChipShellCollection.h>
+
+using namespace chip;
+using namespace chip::Shell;
+using namespace chip::System;
+
+static chip::Shell::Shell sShellDateSubcommands;
+
+int cmd_date_help_iterator(shell_command_t * command, void * arg)
+{
+    streamer_printf(streamer_get(), "  %-15s %s\n\r", command->cmd_name, command->cmd_help);
+    return 0;
+}
+
+int cmd_date_help(int argc, char ** argv)
+{
+    sShellDateSubcommands.ForEachCommand(cmd_date_help_iterator, nullptr);
+    return 0;
+}
+
+int cmd_date_dispatch(int argc, char ** argv)
+{
+    uint16_t year;
+    uint8_t month, dayOfMonth;
+    uint8_t hour, minute, second;
+
+    uint64_t currTimeMS = 0;
+
+    streamer_t * sout = streamer_get();
+
+    if (0 == argc)
+    {
+        if (Layer::GetClock_RealTimeMS(currTimeMS) != CHIP_SYSTEM_NO_ERROR)
+        {
+            streamer_printf(sout, "ERROR: Date/Time was not set\r\n");
+            return CHIP_ERROR_ACCESS_DENIED;
+        }
+
+        SecondsSinceEpochToCalendarTime(currTimeMS / 1000, year, month, dayOfMonth, hour, minute, second);
+        streamer_printf(sout, "%04" PRIu16 "-%02" PRIu8 "-%02" PRIu8 " %02" PRIu8 ":%02" PRIu8 ":%02" PRIu16 "\n\r", year, month,
```

Comment:
Why use a `PRIu16` for seconds at the end ?

Reply:
Oh, it looks like typo - we should use PRIu8 here.